### ISSUE TITLE
How is saving handled

### ISSUE THREAD

**Issue by sunnylawn** [verbb/formie#1738]
### Question

We're building a multipage form with the save button, and I need to highlight to the client that this isn't a feature that is going to allow their users to start their form on one device, and finish on another, but I also wanted to define what sort of timescales/options are invovled on the save function.

Saving the form we can see creates it as an incomplete form in the system with a submission ID, and then we are presuming that the submission ID is matched to a user via a session cookie? Is that the case and if not what is matching them up?

Thanks for you help in advance!

**Comment by engram-design** [comment]
That's correct, we use a session variable to store the submission's ID so that it can be loaded later. It's actually something you get for "free" as it's the mechanism we use to retain a submission across multiple pages with a Page Reload form.

It also doesn't expire (apart from your session ending), so coming back to the form will pick up where you left off. Arguably, this should be configurable.

We didn't opt for more advanced saving options like generating a "share URL" which would be a unique URL to pick up where you left off - purely out of not seeing the need for it. But I'm sure people would like some more options here, we've just not had anyone suggest any different yet!

**Comment by engram-design** [comment]
Saving has been radically revamped in [4.0.0-beta.1](https://github.com/verbb/formie/releases/tag/4.0.0-beta.1), being database-driven (no sessions), a configurable "auto pick up where you left off" setting and proper save handling with a share URL.
